h2o-parsers/h2o-parquet-parser/src/main/java/water/parser/parquet/ParquetParser.java

```diff
@@ -257,15 +263,30 @@ private static ParquetPreviewParseWriter readFirstRecords(ParquetParseSetup init
     }
   }
 
-  private static byte[] roughGuessTypes(MessageType messageType) {
-    byte[] types = new byte[messageType.getPaths().size()];
-    for (int i = 0; i < types.length; i++) {
+  private static byte[] roughGuessTypes(final MessageType messageType, final Optional<String[]> optionalPartitionBy, final String vecKey) {
+    final int partitionedByColumnCount = optionalPartitionBy.isPresent() ? optionalPartitionBy.get().length : 0;
+    final int columnCount = messageType.getPaths().size() + (optionalPartitionBy.isPresent() ? optionalPartitionBy.get().length : 0);
+    byte[] types = new byte[columnCount];
+    for (int i = 0; i < types.length - partitionedByColumnCount; i++) {
       Type parquetType = messageType.getType(i);
       assert parquetType.isPrimitive();
       OriginalType ot = parquetType.getOriginalType();
       PrimitiveType pt = parquetType.asPrimitiveType();
       types[i] = convertType(ot, pt);
     }
+
+    if (optionalPartitionBy.isPresent()) {
+      final String[] partitionByColumns = optionalPartitionBy.get();
+      for (int i = 0; i < partitionByColumns.length; i++) {
+        final String partitionBy = partitionByColumns[i];
+        final Pattern partitionByColumnValue = Pattern.compile(".*" + Pattern.quote(partitionBy) + "=" + "(.*)/.*");
+        final Matcher matcher = partitionByColumnValue.matcher(vecKey);
+        if (matcher.matches()) {
+          types[types.length - partitionedByColumnCount + i] = Vec.T_STR;
```

Comment:
Currently I only use STR, as it's uncertain how the type guessing will work in the end.